src/test/ui/issues/issue-36638.stderr

```diff
@@ -17,6 +17,7 @@ LL | struct Foo<Self>(Self);
    |            ^^^^ unused parameter
    |
    = help: consider removing `Self`, referring to it in a field, or using a marker such as `PhantomData`
+   = help: if you intended `Self` to be a const parameter, use `const Self: usize` instead
```

Comment:
Should we annotate this one as well? I think it might not be super useful to add the help note here.

Reply:
seems fine to me. We already errored above because `Self` is now a valid param name.

I don't think it's that helpful to special case this.